File: Insertion Sort/insertionSort.java
// package Insertion Sort;

public class insertionSort {
  public static void main(String[] args) {

    int[] arr = { 6, 5, 4, 3, 2, 1, 10, 9, 8, 7, 6, 5 };
    insertion_sort(arr);
    for (int num : arr) {
      System.out.print(" " + num);
    }

  }

  static void insertion_sort(int[] arr) {
    int n = arr.length;

    for (int i = 1; i < n; i++) {
      // int j = i;
      while (i > 0 && arr[i - 1] > arr[i]) {
        int temp = arr[i - 1];
        arr[i - 1] = arr[i];
        arr[i] = temp;

        i--;
      }
    }
  }
}

File: Insertion Sort/insertionSort.py
def insertion_sort(arr):
  n = len(arr)
  for i in range(0, n - 1):
    while(i > 0 and arr[i - 1 ] > arr[i]):
      arr[i], arr[i+1] = arr[i+1],arr[i]

def main():
  arr = [5,4,3,2,1,9,8,7,6,5]
  insertion_sort(arr)
  print(arr)

if __name__ == "__main__":
  main()